libbpf-tools/memleak.c

```diff
@@ -395,6 +401,30 @@ int main(int argc, char *argv[])
 
 #ifdef USE_BLAZESYM
 	symbolizer = blazesym_new();
+	if (!symbolizer) {
+		fprintf(stderr, "Failed to load blazesym\n");
+
+		goto cleanup;
+	}
+	print_stack_frames_func = print_stack_frames_by_blazesym;
+#else
+	if (env.kernel_trace) {
+		ksyms = ksyms__load();
+		if (!ksyms) {
+			fprintf(stderr, "Failed to load ksyms\n");
+
+			goto cleanup;
+		}
+		print_stack_frames_func = print_stack_frames_by_ksyms;
+	} else {
+		syms_cache = syms_cache__new(0);
+		if (!syms_cache) {
+			fprintf(stderr, "Failed to create syms_cache\n");
+
+			goto cleanup;
```

Comment:
can you set some non-zero error value before jumping to cleanup (here and above)?

Reply:
Done

Reply:
hmm, not a big deal but I wouldn't have expected a permission error. is the implication that privileges are needed for these initializations?

Reply:
Nice view, my first thought was to use a generic errno since there could be various kind of error from low-level parts.

But now I assume `ENOMEM` would be the better one, because `blazesym_new`/`ksyms__load`/`syms_cache__new` could be treated as a kind of memory allocation failure.

what do you think about `ENOMEM` ? Or do you have another proposal ?

Reply:
I think that will be ok for now since it's somewhat related. let's get this feature merged in and we can change the error later on if we think of something better in the future.